### PR TITLE
[INFRA] Fix internal links implicitly

### LINKED ISSUE #974: links to spec sections sometimes broken when generated from schema
For example, in [the sessions-file section](https://bids-specification.readthedocs.io/en/latest/03-modality-agnostic-files.html#sessions-file), look at the `acq_time` metadata row, and try clicking on the `Units` link.

It'll try to link you to `https://bids-specification.readthedocs.io/02-common-principles.html#units`, which doesn't exist --> it should be `https://bids-specification.readthedocs.io/en/latest/02-common-principles.html#units` (or "stable" instead of the "latest").

I think this may be due to the link being specified as `.html` instead of `.md`

https://github.com/bids-standard/bids-specification/blob/febae4fb9b2c9f055dc1392e0283c85f05618de0/src/schema/objects/columns.yaml#L20

cc @tsalo

### LINKED ISSUE #1011: broken link to licenses appendix
On https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html#dataset_descriptionjson in the License description there is a link to https://bids-specification.readthedocs.io/99-appendices/02-licenses.html which is broken. 

A link that does work is https://bids-specification.readthedocs.io/en/stable/99-appendices/02-licenses.html, but I am not sure how that is supposed to work with the different versions.

### LINKED ISSUE #1034: licence annex link broken
https://bids-specification.readthedocs.io/99-appendices/02-licenses.html is 404 - Not found

### LINKED ISSUE #1095: Issue with documentation links being lost in translation to readthedocs
A number of links in the BIDS specification are not being translated correctly in the readthedocs.

For example, in the License entry in the `dataset_description.json` table in the
[Modality agnostic files](https://bids-specification.readthedocs.io/en/stable/03-modality-agnostic-files.html) section, the link to Appendix II is:

*https://bids-specification.readthedocs.io/99-appendices/02-licenses.html*

but should be:

*https://bids-specification.readthedocs.io/en/stable/99-appendices/02-licenses.html*

### PR DESCRIPTION
**Pull request by effigies**
To avoid some boilerplate, I wanted to see if we could get away with fixing linking without modifying the macro calls at all, from the spec side.

Patch on top of #1056. Might just be a bad idea, but I want to see if it builds.

- closes #1056 
- closes #974 
- closes #1034
- closes #1011
- closes #1095

### CODE REVIEW

File: tools/schemacode/schemacode/render.py [hunk 1]
```diff
@@ -119,6 +138,8 @@ def make_glossary(schema):
         obj_desc = obj_desc.replace("\n\n", "<br>")
         # Otherwise a newline corresponds to a space
         obj_desc = obj_desc.replace("\n", " ")
+        # Spec internal links need to be replaced
+        obj_desc = obj_desc.replace("SPEC_ROOT", get_relpath(src_path))
```
Comment (sappelhoff) [inline tools/schemacode/schemacode/render.py:142]: won't this fail when `src_path` is `None`?

File: tools/mkdocs_macros_bids/macros.py [hunk 2]
```diff
@@ -9,6 +9,13 @@
 
 from examplecode import example
 
+def _get_source_path(level=1):
+    import inspect
+    caller = inspect.currentframe().f_back
+    for _ in range(level):
+        caller = caller.f_back
+    return caller.f_locals["_Context__self"]["page"].file.src_path
```
Comment (sappelhoff) [inline tools/mkdocs_macros_bids/macros.py:17]: could you write a docstr with a quick summary of what this does? to make it easier for future maintainers who may not be fully aware of what is happening here.

File: tools/schemacode/schemacode/render.py [hunk 3]
```diff
@@ -14,6 +15,21 @@
 logging.basicConfig(format="%(asctime)-15s [%(levelname)8s] %(message)s")
 
 
+def get_relpath(src_path):
+    """ Retrieve relative path to the source root from the perspective of a Markdown file
```
Comment (sappelhoff) [inline tools/schemacode/schemacode/render.py:19]:
```suggestion
    """Retrieve relative path to the source root from the perspective of a Markdown file.
```

**Review by sappelhoff — COMMENTED** [review]
Thanks @effigies I like this solution. Looks like black magic to me, but why not :smirk:

File: tools/schemacode/schemacode/render.py (hunk 1 above)
Comment (effigies) [reply tools/schemacode/schemacode/render.py:147]: Indeed. `posixpath.relpath(".", None)` works fine, but `posixpath.dirname(None)` doesn't and I forgot that I added that. Please see https://github.com/bids-standard/bids-specification/pull/1096/commits/c7bd1bcd5d5ea3c0db8314cafea38827fe900526.

File: tools/mkdocs_macros_bids/macros.py (hunk 2 above)
Comment (effigies) [reply tools/mkdocs_macros_bids/macros.py]: Done in https://github.com/bids-standard/bids-specification/pull/1096/commits/1aec282e7f6d8838aecb26f1ab0ed432215b4b41.

**Comment by effigies** [comment]
@sappelhoff I squashed your experiments and my reversions of some changes into a single commit so that it would be easy to rebase. Let me know if you prefer to keep your history.

**Review by sappelhoff — APPROVED** [review]
Great, thanks a lot @effigies.

**Pull request merged**